app/controllers/users_controller.rb

```diff
@@ -46,6 +46,35 @@ def check_account_creation_status
     end
   end
 
+  # Run a set of corrections on a user
+  def check
+    unless logged_in_as_admin?
```

Comment:
@shalott Your last paragraph seems contradictory to me.

My instinct is that these are operations on user accounts and therefore should be in the user model. I am more than happy to split them up in to different fixes. 

Can you clarify please.